ArduPlane/mode.h

```diff
@@ -837,6 +879,8 @@ class ModeTakeoff: public Mode
 
     bool does_auto_throttle() const override { return true; }
 
+    bool requires_home() const override { return true; };
```

Comment:
Why does TAKEOFF mode need home? This is an example question applicable to other modes you've marked as "true".

Reply:
TAKEOFF surprised me. When I tested it, the plane did not take off without home. It just sits armed on the runway drawing 0A. 

This is because takeoff mode relies on having home here in the `update()` call: https://github.com/ArduPilot/ardupilot/blob/20c77ae762951fe4410959b94cfe22f44bf07583/ArduPlane/mode_takeoff.cpp#L74

It would be nice if Takeoff could just climb to the altitude using baro alt above arm height, but that doesn't work yet.

I need to test each of the Q* modes; they are all biased to requiring home right now till I confirm they are safe without it.

Reply:
OK, bad example; TAKEOFF is home-relative, so that one makes sense :-)